kgforge/specializations/stores/bluebrain_nexus.py

```diff
@@ -209,18 +210,23 @@ def _register_one(self, resource: Resource, schema_id: str) -> None:
         try:
             schema = quote_plus(schema_id) if schema_id else "_"
             url_base = f"{self.service.url_resources}/{schema}"
-            url = (
-                f"{url_base}/{quote_plus(data['@id'])}"
-                if hasattr(data, "@id")
-                else url_base
-            )
             params_register = copy.deepcopy(self.service.params.get("register", None))
-            response = requests.post(
-                url,
-                headers=self.service.headers,
-                data=json.dumps(data, ensure_ascii=True),
-                params=params_register,
-            )
+            if resource.has_identifier()[0]:
+                url = f"{url_base}/{quote_plus(resource.get_identifier())}"
+                response = requests.put(
```

Comment:
Maybe you can explain why you use `get.put` instead of `get.post`.

Reply:
PUT verb is used here because there is an id provide when creating the resource. POST is more for collection (i.e you are not targeting specific resource)